### PR TITLE
🚧Create image loader

### LINKED ISSUE #551: 🚧Create image loader
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Create an image loader class.  

**Purpose:**
To give the users the ability to load images manually during runtime. 

**Use Case:**
The loader will return a struct of type `ImageData`.  This will give the users the ability to manually manipulate the image during runtime.  They can then use this `ImageData` to create a new `Texture` object used for rendering.  This will basically give the user the ability to manipulate the image before rendering it.

**`ImageLoader` Features:**
- Loads images by pointing to any file on disk.
- Loads images using content relative paths
  > **Note** This means if a full path is not used, then it is assumed that it is a path to the _**Content/Graphics**_ content directory.  If the path is _**NOT**_ a fully qualified path then an extension is not required due to the path representing a path to the content directory.

**`ImageData` Features:**
Add the following features to the `ImageData` struct
- Add a method to the struct to flip the image horizontally
- Add a method to the struct to flip the image vertically
- Add `bool` property to the struct named `IsFlippedHorizontally`
- Add `bool` property to the struct named `IsFlippedVertically`

### Acceptance Criteria
- [x] `ImageLoader` class created.
  - [x] Loads images by pointing to any file on disk.
  - [x] Loads images using relative content paths
- [x] Features added to the `ImageData` struct
  - [x] Method added to the struct to flip the image horizontally
  - [x] Method added to the struct to flip the image vertically
  - [x] `bool` property added to the struct named `IsFlippedHorizontally`
  - [x] `bool` property added to the struct named `IsFlippedVertically`
- [x] Create an issue to create documentation for the website

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [X] Issue linked to the correct milestone _(if applicable)_.
- [x] Draft pull request created and linked to this issue _(only required with code changes)_.

### Issue Dependencies

_No response_

### Related Work

_No response_

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| Workflow Changes    | `workflow` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>
<h2 style="font-weight:bold" align="center">📋Review Checklist📋</h2>

- [x] The **_head(source)_** branch for this pull request is a **_preview feature_** branch, with the correct naming syntax, in the following example:
  - A **_head(source)_** branch is the branch being merged into a target branch.
    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/feature/<issue-num>-<description>
      Example: preview/feature/123-my-preview-feature
      ```
    </details>

- [x] The **_head(source)_** branch for this pull request is created from a **_preview release_** branch, with the correct naming syntax, in the following example:
  <details closed><summary>Example</summary>

    ``` xml
    Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
    Example: preview/v1.2.3-preview.4
    ```
  </details>

- [x] The **_base(target)_** branch for this pull request is a **_preview release_** branch, with the correct naming syntax, in the following example:
  - A **_base(target)_** branch is the branch that the **_head(source)_** branch is merging into.

    <details closed><summary>Example</summary>

      ``` xml
      Syntax: preview/v<major>.<minor>.<patch>-preview.<prev-num>
      Example: preview/v1.2.3-preview.4
      ```
    </details>

💡For more information on branching, refer to the project's [branching documentation](https://github.com/KinsonDigital/.github/blob/master/docs/BranchingFullGitFlow.md).

- [x] Associated issue exists and is linked to this pull request.
  - One issue per pull request.

- [x] Pull request title matches the title of the associated issue.

- [x] A `preview` label has been added to the pull request.

- [x] The labels attached to this PR match the labels attached to the associated issue.

- [x] Project assigned matches the project of the associated issue.

- [x] Milestone assigned matches the milestone of the associated issue.

- [x] My code follows the coding style of this project.
  - The style is enforced by the **_.editorconfig_** files in the project and displayed as warnings.  If there is an edge case, reach out, and let's discuss it.

- [x] I have manually tested my code changes to the best of my ability.

- [x] All tests passed locally.
  - This is required because unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
  - Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
  - Status checks run for this pull request when merging into the **_preview/v\*.\*.\*-preview.\*_** branch.  These status checks run every time a change is pushed to the pull request.  These checks validate version syntax, tagging, builds, unit tests, and more.

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
> **Warning** Added https://github.com/KinsonDigital/Velaptor/labels/%F0%9F%A7%A8breaking%20changes.
The public fields for the `ImageData` struct have been converted into `readonly` properties.

**Comment by CalvinWilkinson** [comment]
> **Warning** Added https://github.com/KinsonDigital/Velaptor/labels/%F0%9F%A7%A8breaking%20changes
> The `Texture` constructor signature has been changed from `public Texture(string, string)` to `public Texture(string, string, ImageData)`.

An additional constructor has been added though to allow the use of the `ImageData` struct.

File: Testing/VelaptorTests/Content/Caching/TextureCacheTests.cs [hunk 1]
```diff
@@ -253,6 +253,8 @@ public void Ctor_WithNullReactableFactoryParam_ThrowsException()
     public void TotalCachedItems_WhenGettingValue_ReturnsCorrectResult()
     {
         // Arrange
+        this.mockPath.Setup(m => m.IsPathRooted(It.IsAny<string?>())).Returns(true);
+        this.mockPath.Setup(m => m.IsPathRooted(It.IsAny<string>())).Returns(true);
```
Comment (KinsonDigitalAdmin) [inline Testing/VelaptorTests/Content/Caching/TextureCacheTests.cs:257]: These lines are duplicated in several locations in the file.  Remove one of the duplicates.

**Review by KinsonDigitalAdmin — CHANGES_REQUESTED** [review]

**Review by KinsonDigitalAdmin — DISMISSED** [review]

**Comment by CalvinWilkinson** [comment]
> **Note** Added a https://github.com/KinsonDigital/Velaptor/labels/%E2%9C%A8new%20feature to `ImageData`
> Added 2 methods with the name `FlipHorizontally()` and `FlipVertically()` to make it easy to flip images.
> Also added 2 properties with the names `IsFlippedHorizontally` and 'IsFlippedVertically` to make it easy to know if the image is flipped in either direction compared to its original state when loaded.

**Review by KinsonDigitalAdmin — APPROVED** [review]

**Pull request merged**